### DISCUSSION TITLE
Enquiry regarding the installation of "CMethods"

### DISCUSSION THREAD

**Discussion by muhshif** [btschwertfeger/python-cmethods#37] [Q&A]
I think the errors happened in my case were due to the mistakes happened while installing "CMethods".
So, can you please assist me to install the "CMethods" package with proper commands (step-by-step) to be run in pip or conda?

**Comment by btschwertfeger** [comment]
The python-cmethods package is currentlly only available on PyPI. To install the module via pip it is recommended to create and activate a virtual environment first:

For bash/sh/zsh on Linux/MacOS /unix-based systems:

```bash
python3 -m venv venv
source venv/bin/activate
python -m pip install python-cmethods
```

Windows cmd:

```cmd
python3 -m venv venv
call venv\Scripts\activate
python3 -m pip install python-cmethods
```

You can also install the tool from source (the example works for unix-based systems). A note here - cloning the current master branch may contain undocumented features, since only tagged releases are considered as stable and included in the documentation at https://python-cmethods.readthedocs.io/en/stable/

```bash
git clone https://github.com/btschwertfeger/python-cmethods.git
cd python-cmethods
python3 -m venv venv
source venv/bin/activate
python3 -m pip install .
```
